### ISSUE TITLE
Annihilation operator with anticommutation relations

### ISSUE THREAD

**Issue by spmvg** [qutip/qutip#863]
I am working with the Fermi-Hubbard model, so I have terms like
![kinetic term](https://user-images.githubusercontent.com/13852721/40267513-f063b94a-5b5d-11e8-9f1e-c8699eeafff6.gif)
where ![c](https://user-images.githubusercontent.com/13852721/40267302-0020220e-5b5b-11e8-8d1c-91d2bb86c688.gif)  is the (Fermionic) annihilation operator for site _i_. I represent this operator in QuTiP as a tensor product of `qutip.qeye(2)` and `qutip.destroy(2)` but this QuTiP operator does not obey the Fermionic anticommutation relations. Is there a way in QuTiP to enforce this?

In other words, I want ![c](https://user-images.githubusercontent.com/13852721/40267379-7b1f6898-5b5b-11e8-92cb-66bc13933967.gif) to behave like
![eq1](https://user-images.githubusercontent.com/13852721/40267393-e0ab8b92-5b5b-11e8-8806-bd5cabb0bba8.gif)
instead of
![eq2](https://user-images.githubusercontent.com/13852721/40267394-e40c2d50-5b5b-11e8-8ee2-e573b5dc0e29.gif)
like my current QuTiP implementation.

I feel like I might be overseeing something. Thanks in advance!

**Comment by sahmed95** [comment]
Can you please post a code snippet of how you are defining the operators? 

Also take a look at this discussion, 
https://groups.google.com/forum/m/#!topic/qutip/jloEQ4Z3IDE

**Comment by nwlambert** [comment]
You need to enforce the correct commutation relations using something like the Jordan-Wigner transformation

https://en.wikipedia.org/wiki/Jordan%E2%80%93Wigner_transformation

**Comment by nathanshammah** [comment]
Dear @spmvg, It could be interesting to look into implementing something along this line in QuTiP, even a small snippet. Or look into possible integrations with [QuSpin](https://github.com/weinbe58/QuSpin) and [OpenFermion](https://github.com/quantumlib/OpenFermion), which implement fermionic commutation relations, although generally looking at other applications. If you develop on this, please feel free to get in touch. Cheers, Nathan

**Comment by IIAOPSW** [comment]
I think this works:

   c0 = qutip.tensor([qutip.destroy(2), identity(2)]) 
   c1 = qutip.tensor([sigmaz(), qutip.destroy(2)])

More generally c_k should be tensored with k copies of sigmaz() to the left and n-k copies of identity to the right. When operators are applied on vacuum in descending order, by definition there is no fermion to the left of the operator so the sigmaz() is irrelevant. When operators are applied in the opposite order, a negative is picked up. Thus every pair of commutation relations is satisfied. I think that is sufficient.

**Comment by ajgpitch** [comment]
@IIAOPSW answer seems sufficient, so closing
